### ISSUE TITLE
dict.update() with keyword args is not supported

### ISSUE THREAD

**Issue by gvanrossum** [python/mypy#1031]
Example:

```
a = dict(foo=42)
a.update(foo=0)
```

Error:

```
/Users/guido/mypy_tests/mypy_update.py:2: error: Unexpected keyword argument "foo" for "update" of "dict"
```

The work-around is pretty simple: `a.update(dict(foo=0))`.

**Comment by kirbyfan64** [comment]
Isn't this a bug in typeshed?

**Comment by kirbyfan64** [comment]
https://github.com/python/typeshed/blob/master/builtins/3/builtins.pyi#L495

**Comment by gvanrossum** [comment]
Well, I tried to fix it with an overload in typeshed (see attached diff -- I also had to add the same overload to Counter) but it didn't really work. Using the overload with keyword args requires the key type to be a string (but there's no way to express that in the stub), and the actual example was passing two values of different types. This example still elicits an error for the update() but not for the dict() call:
```
a = dict(foo=42, bar=False)
a.update(foo=0, bar=False)
```
I always get
```
mypy_update.py:2: error: No overload variant of "update" of "dict" matches argument types [builtins.int, builtins.bool]
```
Adding a type annotation (`# type: Dict[str, Any]`) to line 1 doesn't help.
[update.diff.txt](https://github.com/JukkaL/mypy/files/48908/update.diff.txt)
(Off-topic: I find it ironic that GitHub doesn't support attaching files whose name ends in .diff or .patch. :-)
